src/modules/blockchainApplication/manage/store/reducer.js

```diff
@@ -44,11 +44,15 @@ export const applications = (state = initialState.applications, { type, app, app
       // In cases where a new node for an existing application is being added,
       // the new node url should be appended to the apis array of the application
       if (app.chainID in state) {
-        state[app.chainID].serviceURLs.push(app.serviceURLs);
-      } else {
-        state[app.chainID] = app;
+        return {
+          ...state,
+          [app.chainID]: {
+            serviceURLs: app.serviceURLs,
```

Comment:
Do we really want to replace the whole application object with just one serviceUrl? Probably this is what we want?
This will keep it.
```
      if (app.chainID in state) {
        return {
          ...state,
          [app.chainID]: {
            ...state[app.chainID], // We still want the rest of the fields of an application, right?
            serviceURLs: app.serviceURLs,
          },
        };
      }
```

But why do we need this if statement at all? couldnt we just replace the whole application object with the new one?
Just having:
`return { ...state, [app.chainID]: app };`

or does `app` in this context only contain `serviceURLs`?

Reply:
true actually I guess we can just replace the whole application